lib/skate/settings/test_group.ex

```diff
@@ -0,0 +1,72 @@
+defmodule Skate.Settings.TestGroup do
+  alias Skate.Settings.Db.TestGroup, as: DbTestGroup
+  alias Skate.Settings.Db.User, as: DbUser
+
+  @type t() :: %__MODULE__{
+          id: integer(),
+          name: String.t(),
+          users: [DbUser.t()]
+        }
+
+  @enforce_keys [:id, :name, :users]
+
+  defstruct [:id, :name, users: []]
+
+  @spec create(String.t()) :: __MODULE__.t()
+  def create(name) do
+    %DbTestGroup{name: name}
+    |> Skate.Repo.insert!()
+    |> Skate.Repo.preload(:users)
+    |> convert_from_db_test_group()
+  end
+
+  @spec get_all() :: [t()]
+  def get_all() do
+    DbTestGroup
+    |> Skate.Repo.all()
+    |> Skate.Repo.preload(:users)
+    |> Enum.map(&convert_from_db_test_group(&1))
+  end
+
+  @spec update(t()) :: t()
+  def update(test_group) do
+    existing_test_group =
+      DbTestGroup |> Skate.Repo.get(test_group.id) |> Skate.Repo.preload(:test_group_users)
+
+    existing_test_group
+    |> DbTestGroup.changeset(
+      test_group
+      |> Map.from_struct()
+      |> Map.put(
+        :test_group_users,
+        Enum.map(test_group.users, fn user ->
+          existing_test_group_user =
+            existing_test_group.test_group_users
+            |> Enum.find(&(&1.user_id == user.id))
```

Comment:
suggestion (non-blocking): Maybe using `MapSet.new/2` to create a set of existing user ids could simplify determining if  a user is part of the existing group with a `member?` check

Reply:
Hmm I'm not sure if there's a good way to use `MapSet` to make this better, since we don't just need the existing user IDs but also the ID of the `test_group_user` corresponding to them. This is important so that we don't just delete and recreate those entries every time we add or remove a different user. Let me know if I'm missing something about your idea though!

Reply:
Ah yeah I see what you mean! Maybe a regular map then? Still a non-blocking suggestion though w/ some shorthand below!
```
# Above the changeset creation
existing_user_id_lookup = Map.new(existing_test_group, & %{&1.user_id, &1.id}

# as part of changeset creation
Enum.map(test_group_users, fn user -> 
    id = Map.get(existing_user_id_lookup, user.id, nil)
    %{id: id, user_id: user.id, test_group_id: existing_test_group.id}
end)
```